Date: 2024-04-15
Author: Bernhard M. Wiedemann <bmwiedemann suse de>

Workaround variations in build results between -j1 and -j4 builds.

Compilation time on 4-core VM increases from 211 to 322 seconds,
while 1-core VM remains at 519 s.
 
This patch applies to go1.15, go1.14 and go1.13 packages.

Index: go/src/make.bash
===================================================================
--- go.orig/src/make.bash
+++ go/src/make.bash
@@ -209,7 +209,7 @@ fi
 # Run dist bootstrap to complete make.bash.
 # Bootstrap installs a proper cmd/dist, built with the new toolchain.
 # Throw ours, built with Go 1.4, away after bootstrap.
-./cmd/dist/dist bootstrap $buildall $vflag $GO_DISTFLAGS "$@"
+taskset 1 ./cmd/dist/dist bootstrap $buildall $vflag $GO_DISTFLAGS "$@"
 rm -f ./cmd/dist/dist
 
 # DO NOT ADD ANY NEW CODE HERE.
